dataland-frontend/src/components/forms/parts/kpiSelection/KPIfieldSet.vue

```diff
@@ -0,0 +1,166 @@
+<template>
+  <div data-test="dataPointToggle" class="form-field vertical-middle">
+    <InputSwitch
+      data-test="dataPointToggleButton"
+      inputId="dataPointIsAvailableSwitch"
+      @click="dataPointAvailableToggle"
+      v-model="dataPointIsAvailable"
+    />
+    <h5 data-test="dataPointToggleTitle" class="ml-2">
+      {{ dataPointIsAvailable ? "Data point is available" : "Data point is not available" }}
+    </h5>
+  </div>
+  <div v-show="dataPointIsAvailable">
+    <div class="form-field" v-if="dataPointIsAvailable">
+      <UploadFormHeader
+        :name="valueType === 'percent' ? `${kpiNameMappings[name]} (%)` : kpiNameMappings[name]"
+        :explanation="kpiInfoMappings[name] ?? ''"
+      />
+      <FormKit
+        :disabled="!dataPointIsAvailable"
+        type="number"
+        name="value"
+        v-model="currentMainValue"
+        validation-label=""
+        :placeholder="valueType === 'percent' ? 'Value %' : 'Value'"
+        step="any"
+        min="0"
+        :validation="valueType === 'percent' ? 'number|between:0,100' : 'number'"
+        :inner-class="{
+          short: true,
+        }"
+      />
+    </div>
+
+    <div class="form-field">
+      <FormKit type="group" v-if="dataPointIsAvailable" name="dataSource">
+        <h4 class="mt-0">Data source</h4>
+        <div class="next-to-each-other">
+          <div class="flex-1">
+            <UploadFormHeader :name="kpiNameMappings.report ?? ''" :explanation="kpiInfoMappings.report ?? ''" />
+            <FormKit
+              type="select"
+              name="report"
+              v-model="currentReportValue"
+              :disabled="!dataPointIsAvailable"
+              placeholder="Select a report"
+              :options="['None...', ...reportsName]"
+            />
+          </div>
+          <div>
+            <UploadFormHeader :name="kpiNameMappings.page ?? ''" :explanation="kpiInfoMappings.page ?? ''" />
+            <FormKit
+              outer-class="w-100"
+              :disabled="!dataPointIsAvailable"
+              v-model="currentPageValue"
+              type="number"
+              name="page"
+              placeholder="Page"
+              validation-label="Page"
+            />
```

Comment:
Just an idea: We could disable the "Page" field if no report is selected.  Then as soon as a report is selected, we display it.
What do you think?

Reply:
Put into backlog for further refinement.